app/assets/javascripts/components/overview/details.jsx

```diff
@@ -86,6 +86,31 @@ const Details = createReactClass({
     return true;
   },
 
+  handleWikiChange(wiki) {
+    wiki = wiki.value;
+    const prev_wiki = { ...this.props.course.home_wiki };
+    const updatedCourse = this.props.course;
+    updatedCourse.home_wiki.language = wiki.language;
+    updatedCourse.home_wiki.project = wiki.project;
+    if (!updatedCourse.wikis.filter(w => w.language === wiki.language && w.project === wiki.project).length) {
+      updatedCourse.wikis = [wiki, ...updatedCourse.wikis];
+    } else {
+      updatedCourse.wikis = [...updatedCourse.wikis];
+    }
+    // Remove automatically added previous wiki
```

Comment:
It looks like this is duplicated from CourseForm. Maybe the pairing of home_wiki and tracked_wikis can be extracted into one component that can be used in both Details and CourseForm.